### PR TITLE
Updates HOMME Perlmutter machine files.

### PR DESCRIPTION
**Pull request by jasonltorchinsky**
(Summary Below)
Functional update to HOMME Perlmutter 'nocuda' machine files, which previously caused errors when running DCMIP2016 Test 2 on Perlmutter. Tools to enable building all DCMIP test cases for standalone HOMME on Perlmutter, and to run DCMIP2016 Test 2 on Perlmutter.

* Updates HOMME Perlmutter 'nocuda' machine files to enable PIO.
* Cleans namelists for DCMIP2016 Test 2.
* Adds NERSC jobscript for DCMIP2016 Test 2.
* Adds NERSC build scripts for standalone HOMME.

[NML] Formatting only on DCMIP 2016 Test 2 namelists.
[BFB]

### CODE REVIEW

**Comment by rljacob** [comment]
@oksanaguba can you look at this?

**Comment by oksanaguba** [comment]
@rljacob i can integrate it, Mark probably still need to approve it?

File: components/homme/compile_scripts/nersc/build-homme-nersc.sh [hunk 1]
```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
```
Comment (oksanaguba) [inline components/homme/compile_scripts/nersc/build-homme-nersc.sh:1]: does this file need to be a part of homme repo?

File: components/homme/compile_scripts/nersc/get_case_env.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (oksanaguba) [inline components/homme/compile_scripts/nersc/get_case_env.sh:1]: does this file need to be a part of homme repo?

File: components/homme/dcmip_tests/dcmip2016_test2_tropical_cyclone/theta-l/namelist-r100.nl [hunk 3]
```diff
@@ -1,51 +1,51 @@
 !
-! theta: namelist for dcmip2016 test2: tropical cyclone
```
Comment (oksanaguba) [inline components/homme/dcmip_tests/dcmip2016_test2_tropical_cyclone/theta-l/namelist-r100.nl:2]: For the changes in *nl files, there are too many formatting changes to evaluate by eye. I am not sure those runs are used for, say, energy cascades or anything else, so i am not sure how important that those files are not being modified (that is, that params in the namelists are not modified). Maybe @mt5555 has an opinion.

**Comment by oksanaguba** [comment]
Regarding 
```
[NML] Formatting only on DCMIP 2016 Test 2 namelists.
```
-- just in case you are curious, the namelists prob are not used in nightlies, so we won't see namelist changes on cdash, but also i do not think homme suite is capable to report namelist changes (also, if it only formatting, there won't be NML cdash warnings).

File: components/homme/dcmip_tests/dcmip2016_test2_tropical_cyclone/theta-l/namelist-r100.nl (hunk 3 above)
Comment (rljacob) [reply components/homme/dcmip_tests/dcmip2016_test2_tropical_cyclone/theta-l/namelist-r100.nl:2]: If you go to the very top of this page, the gear icon has a "hide whitespace" option.  There's still a lot of changes from changing comments to mixed case.

**Review by ndkeen — APPROVED** [review]
Missed I was tagged to review -- I don't have strong opinions here. Can test something if needed? Otherwise, feel free to do what's best. Note that those files were created before we had the more official machines names of `pm-cpu` and `pm-gpu`.  And we also have compiler names (which I'm not a fan of) of gnu, gnugpu, nvidia, nvidiagpu.  So you could probably find better filenames and remove the "nocuda" concept

**Comment by oksanaguba** [comment]
i am not sure about the status of this PR, @mt5555 ?

**Comment by mt5555** [comment]
I'll merge this soon. it's the first step in a set of PRs updating HOMME DCMIP tests cases with idealized physics. It's currently delayed due to other priorties.

**Comment by rljacob** [comment]
@mt5555 can you merge this without those other PRs?

**Comment by rljacob** [comment]
I'm going to wait for the homme suite to start running on perlmutter before merging this.

**Pull request merged**